src/components/Table/EditColumnsButton.tsx

```diff
@@ -0,0 +1,85 @@
+import { Dispatch, SetStateAction, useCallback, useEffect, useMemo, useRef, useState } from "react";
+import { useMenuTrigger } from "react-aria";
+import { useMenuTriggerState } from "react-stately";
+import { CheckboxGroup, CheckboxGroupItemOption } from "src/inputs";
+import { Css } from "../../Css";
+import { useTestIds } from "../../utils";
+import { Button } from "../Button";
+import { isIconButton, isTextButton, OverlayTrigger, OverlayTriggerProps } from "../internal/OverlayTrigger";
+import { GridColumn, Kinded } from "./GridTable";
+
+interface EditColumnsButtonProps<R extends Kinded, S>
+  extends Pick<OverlayTriggerProps, "trigger" | "placement" | "disabled" | "tooltip"> {
+  columns: GridColumn<R, S>[];
+  setColumns: Dispatch<SetStateAction<GridColumn<R, S>[]>>;
+  title?: string;
+  // for storybook purposes
+  defaultOpen?: boolean;
+}
+
+export function EditColumnsButton<R extends Kinded, S = {}>(props: EditColumnsButtonProps<R, S>) {
+  const { defaultOpen, disabled, columns, setColumns, trigger, title } = props;
+  const state = useMenuTriggerState({ isOpen: defaultOpen });
+  const buttonRef = useRef<HTMLButtonElement>(null);
+  const { menuTriggerProps } = useMenuTrigger({ isDisabled: !!disabled }, state, buttonRef);
+  const tid = useTestIds(
+    props,
+    isTextButton(trigger) ? trigger.label : isIconButton(trigger) ? trigger.icon : trigger.name,
+  );
+
+  const { options, selectedColumns } = useMemo(() => {
+    return columns.reduce(
+      (acc, column) => {
+        // Only include options that can be hidden and have the `name` property defined.
+        if (!column.canHide) return acc;
+        if (!column.name || column.name.length === 0) {
+          console.warn("Column is missing 'name' property required by the Edit Columns button", column);
+          return acc;
+        }
+
+        // If currently visible, then add to selectedColumns
+        if (column.canHide && column.visible) {
+          acc.selectedColumns.push(column.name);
+        }
+
+        // Add current column as an option
+        return { ...acc, options: acc.options.concat({ label: column.name!, value: column.name! }) };
+      },
+      { options: [] as CheckboxGroupItemOption[], selectedColumns: [] as string[] },
+    );
+  }, [columns]);
+
+  const [selectedValues, setSelectedValues] = useState<string[]>(selectedColumns);
+
+  const clearSelections = useCallback(() => {
+    setSelectedValues([]);
+  }, [columns]);
```

Comment:
**nit (non-blocking):** We should try to recognize when a function would benefit or not benefit from being memoized. There is so little logic in this callback now that I don't think it needs to be memoized, and perhaps doesn't even need a defined function as it is only used in one place. We could just do this inline for the "Clear Selections" button. 
If we do decide to keep `useCallback`, there is no reason to any longer have `columns` in the dependency array, so let's at least remove that.